flipkart_scraper.py

```diff
@@ -0,0 +1,133 @@
+        link=container.findAll("a",{"class":"Zhf2z-"})
+        for nextlinks in link:
+          orglink=nextlinks.get('href')
+          #Found the links for the targeted product
+          real="https://www.flipkart.com"+orglink
+          linkss=linkss+[real]
+          print(linkss)
+        maindata=data.findAll("a",{"class":"_2cLu-l"})
+        mt=maindata[0].text
+        subdata=data.findAll("div",{"class":"_1rcHFq"})
+        sb=subdata[0].text
+        #Found title
+        title=title+[mt]
+        rating=data.findAll("div",{"class":"niH0FQ _36Fcw_"})
+        #Found ratings
+        try:
+          oo=rating[0].span.text
+          ratings=ratings+[oo]
+        except:
+          ratings=ratings+["No rating"]
+        nextcontainers = data.findAll("div",{"class":"_1vC4OE"})
+        realprice=nextcontainers[0].text
+        #Found price
+        amount=amount+[realprice]
+  return (linkss, title, ratings, amount, len(amount)) 
+
+def scrape_page(urlss,limit):
+  '''The get_urls return the url's list to the scrape_page() function
+  This function contains the type1 script of the scraper, which is one of the 
+  pattern in which the details of the products are arranged'''
+  title=[]
+  amount=[]
+  linkss=[]
+  ratings=[]
+  #only use n-1 pages for removing out of index error
+  for i in range(1,limit-1):  
+    myurl= urlss[i]
+    ucli= ureq(myurl)
+    #reads the webpage and transfers to a variable
+    page_html = ucli.read() 
+    ucli.close()
+    page_soup = soup(page_html,"html.parser")
+    containers = page_soup.findAll("div",{"class":"bhgxx2 col-12-12"})
+    #After fixing the whole target, the scraper moves towards the content using findAll function 
+    for container in containers:
+      datas=container.findAll("a",{"class":"_31qSD5"})
+      link=container.findAll("div",{"class":"_1UoZlX"})
+      for nextlinks in link:
+        orglink=nextlinks.a.get('href')
+        #Found the links for the targeted product
+        real="https://www.flipkart.com"+orglink
+        linkss=linkss+[real]
+      for data in datas:
+        nextcontainers = data.findAll("div",{"class":"_1uv9Cb"})
+        for cont in nextcontainers:
+          realprice=cont.div.text
+          #Found price
+          amount=amount+[realprice]
+        maindata=data.findAll("div",{"class":"_1-2Iqu row"})
+        for headdata in maindata:
+          head=headdata.findAll("div",{"class":"col col-7-12"})
+          mt=head[0].div.text
+          #Found title
+          title=title+[mt]
+          #Found ratings
+          rating=headdata.findAll("div",{"class":"niH0FQ"})
+          try:
+            oo=rating[0].span.text
+            ratings=ratings+[oo]
+          except:
+            ratings=ratings+["No rating"]
+          
+  #for different type of page  
+  if not title:  
+   answer = type2(urlss,limit)
+   return answer
+  else:
+   '''And finally the function returns the list of links,ratings,amount,number of products.'''
```

Comment:
This can be a simple comment because docstrings are only used to describe functions, classes, etc :)